File: posawesome/posawesome/api/pos_profile.py
import frappe

@frappe.whitelist()
def get_posa_invoice_naming_series():
	series = frappe.get_meta("Sales Invoice").get_field("naming_series").options
	frappe.db.set_value(
		"Custom Field",
		"POS Profile-posa_invoice_naming_series",
		"options",
		series
	)

	frappe.clear_cache()
	return {"options": series}

File: posawesome/posawesome/api/pos_profile.js
frappe.ui.form.on('POS Profile', {
	setup: function (frm) {
		frm.set_query("posa_cash_mode_of_payment", function (doc) {
			return {
				filters: { 'type': 'Cash' }
			};
		});
	},
	refresh: async function (frm) {

		await frappe.xcall("posawesome.posawesome.api.pos_profile.get_posa_invoice_naming_series").then(r => {
			console.log("POS Invoice Naming Series:", r.options);
			frm.set_df_property('posa_invoice_naming_series', 'options', r.options);
			frm.refresh_field('posa_invoice_naming_series');
		}).catch(e => {
			console.error(e);
			frappe.msgprint(__("Failed to fetch POS Invoice Naming Series"), __("Error"));
		});
	}
});